hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/initialization/BlocklistAccountCreator.java

```diff
@@ -0,0 +1,251 @@
+            final var account = blockedAccountWith(blockedInfo);
+            accounts.put(newId, account);
+            blockedAccountsCreated.add(account);
+            aliasManager.link(blockedInfo.evmAddress, EntityNum.fromAccountId(newId));
+        }
+    }
+
+    private List<String> readPrivateKeyBlocklist(String fileName) {
+        final var inputStream = getClass().getClassLoader().getResourceAsStream(fileName);
+        final var reader = new BufferedReader(new InputStreamReader(inputStream));
+        return reader.lines().toList();
+    }
+
+    /**
+     * Parses a line from the blocklist file and returns blocked account info record.
+     *
+     * The line should have the following format:
+     * <private key>,<memo>
+     *     where <private key> is a hex-encoded private key
+     *     and <memo> is a memo for the blocked account
+     *     and both values are comma-separated.
+     *
+     * The resulting blocked account info record contains the EVM address derived from the private key, and the memo.
+     *
+     * @param line line from the blocklist file
+     * @param columnCount number of comma-separated values in a line
+     * @return blocked account info record
+     */
+    private BlockedInfo parseCSVLine(String line, int columnCount) {
+        final var parts = line.split(",");
+        if (parts.length != columnCount) {
+            throw new IllegalArgumentException("Invalid line in blocklist file: " + line);
+        }
+
+        final var privateKeyBytes = Hex.decode(parts[0]);
+        final var publicKeyBytes = ecdsaPrivateToPublicKey(privateKeyBytes);
+        final var evmAddressBytes = EthSigsUtils.recoverAddressFromPubKey(publicKeyBytes);
+        return new BlockedInfo(ByteString.copyFrom(evmAddressBytes), parts[1]);
+    }
+
+    /**
+     * Creates a blocked Hedera account with the given memo and EVM address.
+     * A blocked account has receiverSigRequired flag set to true, key set to the genesis key, and balance set to 0.
+     *
+     * @param blockedInfo record containing EVM address and memo for the blocked account
+     * @return a Hedera account with the given memo and EVM address
+     */
+    private HederaAccount blockedAccountWith(BlockedInfo blockedInfo) {
+        final var expiry = properties.getLongProperty(BOOTSTRAP_SYSTEM_ENTITY_EXPIRY);
+        final var account = new HederaAccountCustomizer()
+                .isReceiverSigRequired(true)
+                .isDeclinedReward(true)
+                .isDeleted(false)
+                .expiry(expiry)
+                .memo(blockedInfo.memo)
+                .isSmartContract(false)
+                .key(getGenesisKey())
+                .autoRenewPeriod(expiry)
+                .alias(blockedInfo.evmAddress)
+                .customizing(accountSupplier.get());
+        try {
+            account.setBalance(ZERO_BALANCE);
+        } catch (final NegativeAccountBalanceException e) {
+            throw new IllegalStateException(e);
```

Comment:
Q: How is this exception possible?

Reply:
It the current case it is not possible, it's just that `account.setBalance` can throw a checked exception. I see that there is also `account.setBalanceUnchecked` method that throws an unchecked exception that might be a better option here, but giving it a second thought the default value of an account's balance should be zero, so setting it explicitly seems redundant. I will remove this altogether.